### PR TITLE
Migrating from `util::Optional` to `std::optional` since Realm Core is moving that way

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

The Realm Core team wishes to stop using `util::Optional` and start using `std::optional`. Currently `util::Optional` is an alias but will eventually disappear. Together with https://github.com/realm/realm-core/pull/7284 we are preparing for the future.

Not sure why `package-lock.json` needs to be updated.

## ☑️ ToDos

* [x] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [ ] 🚦 Tests
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@
 <!-- * Either mention core version or upgrade -->
 <!-- * Using Realm Core vX.Y.Z -->
 <!-- * Upgraded Realm Core from vX.Y.Z to vA.B.C -->
+* Migrated bingen from `util::Optional` to `std::optional`
```
Comment (kneth) [inline CHANGELOG.md:22]:
```suggestion
* Migrated bindgen from `util::Optional` to `std::optional`
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -19,6 +19,7 @@
 <!-- * Either mention core version or upgrade -->
 <!-- * Using Realm Core vX.Y.Z -->
 <!-- * Upgraded Realm Core from vX.Y.Z to vA.B.C -->
```
Comment (elle-j) [inline CHANGELOG.md:21]:
```suggestion
```

File: package-lock.json
Comment (elle-j) [inline package-lock.json:1]: This one changed because of `commander` and `@commander-js/extra-typings` switching versions in bindgen to `v11`.

**Review by elle-j — APPROVED** [review]
Future, here we come 🙂

File: package-lock.json
Comment (kraenhansen) [reply package-lock.json:1]: Yep 👍 It's expected and will also happen when we pull in https://github.com/realm/realm-core/pull/7299 😬

**Review by kraenhansen — APPROVED** [review]
If it builds and passes, I'm happy 😊

**Pull request merged**